### PR TITLE
Converted Archer login from REST to SOAP and added logout

### PR DESCRIPTION
**Pull request by sagylell**
fixes: https://github.com/demisto/etc/issues/10352

### CODE REVIEW

File: Integrations/integration-ArcherRSA.yml [hunk 1]
```diff
@@ -967,46 +995,54 @@ script:
...
-        case 'fetch-incidents':
-            return fetchIncidents(getAuthToken(), args);
-        default:
-            break;
+    var token = null;
+    try {
+        token = getAuthToken();
+        switch (command) {
+            case 'test-module':
+                return !!token ? 'ok' : 'not-ok';
```
Comment (anara123) [inline Integrations/integration-ArcherRSA.yml]: if you return here then you will never reach `finally` block and logout.

File: Integrations/integration-ArcherRSA.yml [hunk 2]
```diff
@@ -967,46 +995,54 @@ script:
...
-
-        if (result.IsSuccessful != true) {
-            throw 'Login failed. Result = ' + JSON.stringify(result);
+        var soap = createUserSessionFromInstanceSoapRequest(params.instanceName,
+                                                            params.credentials.identifier,
+                                                            params.credentials.password);
+        var SOAPAction = 'http://archer-tech.com/webservices/CreateUserSessionFromInstance';
+        var response = JSON.parse(x2j(sendSoapRequest('login', soap, SOAPAction)));
+        if (response === null || typeof response !== 'object' || response === undefined) {
+            throw 'Login failed. Result = ' + response;
```
Comment (anara123) [inline Integrations/integration-ArcherRSA.yml]: most probably if you reached this line, your response will not contain nothing (null). You should print the raw response of the soapRequest

**Comment by anara123** [comment]
Your build is still failing. Merge from master and push

**Comment by aviadl** [comment]
DO NOT MERGE WITHOUT MY APPROVAL please

**Pull request merged**